open_spiel/games/dark_hex.cc

```diff
@@ -323,5 +325,63 @@ std::vector<int> DarkHexGame::ObservationTensorShape() const {
   }
 }
 
+CellState ImperfectRecallDarkHexState::board(Player player, int row, int col) const {
+  // Returns the value of the cell in the given players board and the 2d coordinates.
+  const auto& player_view = (player == 0 ? black_view_ : white_view_);
+  return player_view[row * num_cols() + col];
+}
+
+int ImperfectRecallDarkHexState::observation_plane(Player player, int r, int c) const {
+  int plane = -1;
+  int cell = static_cast<int>(board(player, r, c));
+  if (cell > 0 && cell <= -kMinValueCellState) {
+    plane = 0;
+  } else if (cell < 0 && cell >= kMinValueCellState) {
+    plane = 1;
+  } else if (cell == 0) {
+    plane = 2;
+  } else {
+    std::cerr << "Invalid character on board: " << cell
+                << std::endl;
+    SpielFatalError("Invalid cell state");
+  }
+
+  return plane;
+}
+
+std::vector<int> ImperfectRecallDarkHexGame::InformationStateTensorShape() const {
+  return {3, num_rows(), num_cols()}; // 3 planes: black, white, empty
+}
+
+void ImperfectRecallDarkHexState::InformationStateTensor(Player player,
+                                          absl::Span<float> values) const {
+  SPIEL_CHECK_GE(player, 0);
+  SPIEL_CHECK_LT(player, num_players_);
+
+  SPIEL_CHECK_EQ(values.size(), 3 * num_rows() * num_cols());
+  std::fill(values.begin(), values.end(), 0.);
+
+  // Treat `values` as a 3-d tensor.
+  TensorView<3> view(values, {3, num_rows(), num_cols()}, true);
+
+  for (int r = 0; r < num_rows(); r++) {
+    for (int c = 0; c < num_cols(); c++) {
+      int plane = observation_plane(player, r, c);
+      SPIEL_CHECK_TRUE(plane >= 0 && plane < 3);
+      view[{plane, r, c}] = 1.0;
+    }
+  }
+}
+
+void ImperfectRecallDarkHexState::ObservationTensor(Player player,
+                                     absl::Span<float> values) const {
+  // Same as the InformationStateTensor.
+  InformationStateTensor(player, values);
```

Comment:
Not sure if this is legal for OS, but it is unnecessary to repeat the code I think.